hclee/process_all_in_one.py

```diff
@@ -0,0 +1,349 @@
+
+                        # Solve PnP
+                        success, rot_vec, trans_vec = cv2.solvePnP(face_3d, face_2d, cam_matrix, dist_matrix)
+
+                        # Get rotational matrix
+                        rmat, jac = cv2.Rodrigues(rot_vec)
+
+                        # Get angles
+                        angles, mtxR, mtxQ, Qx, Qy, Qz = cv2.RQDecomp3x3(rmat)
+                        arr['roll'][i]  = angles[0]
+                        arr['pitch'][i] = angles[1]
+                        arr['yaw'][i]   = angles[2]
+                    ## Head pose ##########################
+                    i+=1
+
+                    #iris point를 추출하고 리스트에 저장
+                    irisPoints = []
+                    if results.multi_face_landmarks:
+                        for face_landmarks in results.multi_face_landmarks:
+                            for start,end in irisIndex:
+                                irisx = face_landmarks.landmark[start].x
+                                irisy = face_landmarks.landmark[start].y
+                                
+                                irisxScale = int(irisx * 1920)
+                                irisyScale = int(irisy * 1080)
+                                
+                                irisPoints.append([irisxScale, irisyScale])
+                    
+                    #왼쪽 눈의 좌우, 오른쪽 눈의 좌우를 활용하여 두 눈 사이의 거리를 계산   
+                    left1x, left1y = irisPoints[5]
+                    left2x, left2y = irisPoints[6]
+                    
+                    right1x, right1y = irisPoints[2]
+                    right2x, right2y = irisPoints[4]
+                    
+                    leftIrisSizePix = ( ((left1x-left2x) ** 2) + ((left1y-left2y) ** 2) ) ** 0.5
+                    rightIrisSizePix = ( ((right1x-right2x) ** 2) + ((right1y-right2y) ** 2) ) ** 0.5
+                    leftIrisSize = leftIrisSizePix * pixSize
+                    rightIrisSize = rightIrisSizePix * pixSize
+                    
+                    distanceLeft = ( Focal * irisRealSize ) / (leftIrisSize) 
+                    distanceRight = ( Focal * irisRealSize ) / (rightIrisSize)
+                    
+                    distance = ( distanceLeft + distanceRight ) / 2 * 100   #cm로 변환
+                    
+                    resultDistance.append(int(distance))
+                except:     #iris를 못 찾을 경우 이전 값을 그대로 사용
+                    resultDistance.append(int(distance))
+                    continue
+            
+            video.release()
+
+            # Post process 
+            factor = 360*10 # Temporary...
+            process_head(arr, factor)
+
+
+            ### Save head pose 
+            with open(saveName_head, "w") as f:
+                f.write("[head angle] roll      pitch      yaw\n")
+                for roll, pitch, yaw in arr:
+                    f.write(f"{roll:.4f}, {pitch:.4f}, {yaw:.4f}\n")
```

Comment:
This comes to personal taste, but I think python unpacking like this is both unsafe and hard to maintain. Say someone decide to add some field in this 'arr', or change the order to 'yaw, pitch, roll', it would require to change every subsequent code that unpack the array
Should be explicitly unpacked:
```
for headpose in arr:
  f.write(f"{headpose['roll']:.4f}, {headpose['pitch']:.4f}, {headpose['yaw']:.4f}\n") 
```